tests/validator20/validate_spec_test.py

```diff
@@ -423,3 +424,11 @@ def test_highlight_inconsistent_schema_object_validation(minimal_swagger_dict, s
     minimal_swagger_dict.update(swagger_dict_override)
     with pytest.raises(SwaggerValidationError):
         validate_spec(minimal_swagger_dict)
+
+
+def test_validation_performance(minimal_swagger_dict):
+    start = time.time()
+    for _ in range(20):
+        validate_spec(minimal_swagger_dict)
+
+    assert time.time() - start < 1
```

Comment:
Not very comfortable with this test.
I understand the spirit of it, but
 * what if we're running tests on a very very very slow machine? we might take more than a second even with the optimisation in place
 * what if we're running on a very very very fast machine? we might take less then a second even without the optimisation you put in place.

I would rather recommend to test that if you're calling `validate_spec` more than once (twice) then the uncached `read_file` method is called only once (hint. `mock.Mock(..., wraps=...)` )